src/actors/hcd/neutral_fueling_actor.jl

```diff
@@ -0,0 +1,417 @@
+    amin = 0.5.*(dd.equilibrium.time_slice[].profiles_1d.r_outboard[end]-
+                 dd.equilibrium.time_slice[].profiles_1d.r_inboard[end])
+    Rmaj = 0.5.*(dd.equilibrium.time_slice[].profiles_1d.r_outboard[end]+
+                 dd.equilibrium.time_slice[].profiles_1d.r_inboard[end])
+    raneut = sqrt.(kappa).*amin
+    rhoa = dd.equilibrium.time_slice[].profiles_1d.rho_tor[end]
+    ratef = raneut / rhoa
+    eionr = ratef .* eir(Te)
+    eirate = eir(dd.core_profiles.profiles_1d[].electrons.temperature/1e3).*ne
+
+    vth = 1e2 .* sqrt.(2*1.6e-19*(Te*1e3) ./ (mi*1.6e-27))
+ 
+    a1 = ne .* eionr .+ ni .* cxr(Te/mi)
+    b1 = ni .* cxr(Te/mi) 
+    vwall  = par.vwall .* sqrt.(2.0*1.6e-19*twall ./ (mi*1.67e-27))
+    
+    flux1 = IMAS.trapz(surface, ni)/taupin
+    swall = 2.0*flux1/vwall
+
+    theta = collect(0:0.05*(pi/2):pi/2)
+    
+    r = rho*1e2*rhoa
+    ra = deepcopy(r)
+    nr = length(r)
+    nra = length(ra)
+    nt = length(theta)
+    
+    odelr = 1.0/(r[2] - r[1])
+    
+    sint = sin.(theta)
+    
+    k11 = zeros(nr, nr)
+    atrr01 = zeros(nr, nt)
+    atar01 = zeros(nr, nt)
+
+    for i in 1:nr
+        k11[i, i] = 0.0
+        
+        for k in 1:nt
+            r0 = r[i] * sint[k]
+            atrr01[i:nr, k] .= get_At(r[i:nr], nr - i + 1, r0, odelr,ra,nra, a1)
+            atar01[i, k] = atrr01[nr, k]
+        end
+    
+        if i == 1
+            for kk in 2:nt
+                for ii in 1:nr
+                atrr01[ii, kk] = atrr01[ii, 1]
+                end
+                atar01[1, kk] = atrr01[nr, kk]
+            end
+        end
+    end
+
+    
+    G(x,n) = 2.0/sqrt(π)*QuadGK.quadgk(η-> η.^n .* exp.(-η.^2 .- x ./ η), 0., Inf)[1]
+    G1h = zeros(nr, nt)
+    Ap = zeros(nr, nt)
+    Am = zeros(nr,nt)
+    h = zeros(nr)
+    for (ir,rr) in enumerate(rho)
+        for (it,tt) in enumerate(theta)
+            Ap[ir,it] = abs.(atrr01[ir,it] + atar01[ir,it])
+            Am[ir,it] = abs.(atrr01[ir,it] - atar01[ir,it])
+            G1h[ir,it] = G(Ap[ir,it]/vwall,1)+G(Am[ir,it]/vwall,1)
+        end
+    end
+
+
+    for (ir,rr) in enumerate(rho)
+        h[ir] = swall*IMAS.trapz(theta,G1h[ir,:])
+    end
+
+    k11 = get_K_matrix(atrr01,vth,r,nr,theta,nt,sint)
+    k11 = b1./vth./sqrt(π) .* k11 
+
+    pn11 = nuslv1(h,k11, nr)
+
+    cm3_to_m3 = 1e6
+    fn11 = pn11 * cm3_to_m3
+    sione = fn11.*eirate
+
+    correction = (trapz(volume,ni)/taupin)/trapz(volume,sione/cm3_to_m3)
+ 
+    return sione*correction,fn11*correction
+    
+end
+
+"""
+    ActorNeutralFueling(dd::IMAS.dd, act::ParametersAllActors; kw...)
+
+Estimates the neutral fueling deposition
+"""
+function ActorNeutralFueling(dd::IMAS.dd, act::ParametersAllActors; kw...)
+    actor = ActorNeutralFueling(dd, act.ActorNeutralFueling; kw...)
+    step(actor)
+    finalize(actor)
+    return actor
+end
+
+
+function _step(actor::ActorNeutralFueling)
+    dd = actor.dd
+    par = actor.par
+
+    if par.model == :neucg
```

Comment:
if model is not `:neucg` this will throw an error

Reply:
fixed